lib/terminal-element.js

```diff
@@ -0,0 +1,101 @@
+/** @babel */
+
+import { CompositeDisposable } from 'atom';
+import ResizeObserver from 'resize-observer-polyfill';
+
+class TerminalElement extends HTMLElement {
+
+  initialize(session) {
```

Comment:
Should the TerminalElement own the Xterm UI or does that complicate things too much?  If TerminalSession exposes the pty object, TerminalElement could attach to it once initialized.

Reply:
So, I've been going back and forth on this. Ultimately, I put it in the model because I consider the base Terminal instance to be the model for the view inserted into the DOM with `open()`. However, it doesn't feel clean either way, since both the model and view need to interact with the Terminal instance for resizing, UI commands (i.e. clearing, copying, pasting), etc.

The other thing is, it feels like the view should be reaching into the model, not the model reaching into the view... I'm not seeing a lot of good pattern examples out there in existing addons I've looked at.